znvis/particle/vector_field.py

```
@@ -116,9 +116,11 @@ def construct_mesh_list(self):
                 n_time_steps = 1
                 self.position = self.position[np.newaxis, :, :]
                 self.direction = self.direction[np.newaxis, :, :]
-
-        except ValueError:
-            raise ValueError("There is no data for this vector field.")
+        # NOTE: The earlier exception was never called, so I'm removing it.
+        except IndexError:
+            raise IndexError("The provided data has an incompatible shape.")
```

Ruff (0.13.3)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)